fang/src/asynk/backend_sqlx.rs

```diff
@@ -0,0 +1,476 @@
+use chrono::{DateTime, Duration, Utc};
+use sha2::Digest;
+use sha2::Sha256;
+use sqlx::any::AnyQueryResult;
+use sqlx::database::HasArguments;
+use sqlx::Database;
+use sqlx::Encode;
+use sqlx::Executor;
+use sqlx::FromRow;
+use sqlx::IntoArguments;
+use sqlx::Pool;
+use sqlx::Type;
+use std::fmt::Debug;
+use typed_builder::TypedBuilder;
+use uuid::Uuid;
+
+#[cfg(feature = "asynk-postgres")]
+mod postgres;
+#[cfg(feature = "asynk-postgres")]
+use self::postgres::BackendSqlXPg;
+
+#[cfg(feature = "asynk-sqlite")]
+mod sqlite;
+#[cfg(feature = "asynk-sqlite")]
+use self::sqlite::BackendSqlXSQLite;
+#[cfg(feature = "asynk-mysql")]
+mod mysql;
+#[cfg(feature = "asynk-mysql")]
+use self::mysql::BackendSqlXMySQL;
+
+#[derive(Debug, Clone)]
+pub(crate) enum BackendSqlX {
+    #[cfg(feature = "asynk-postgres")]
+    Pg,
+
+    #[cfg(feature = "asynk-sqlite")]
+    Sqlite,
+
+    #[cfg(feature = "asynk-mysql")]
+    MySql,
+
+    #[cfg(not(any(
+        feature = "asynk-postgres",
+        feature = "asynk-sqlite",
+        feature = "asynk-mysql"
+    )))]
+    #[allow(dead_code)]
+    Dummy,
```

Comment:
In my attempt to remove the `backend` field from `AsyncQueue`, since it was redundant, I had to add this `Dummy` variant to `BackendSqlx` to ensure this type is always inhabited to convince the compiler there is no dead code.

However, I think this is a bit messy and would appreciate if anyone comes up with an alternative. Reverting my last commits may also be a preferable path.

Reply:
no problem, i have prepared a better fix

Reply:
[use variant NoBackend instead Option](https://github.com/ayrat555/fang/pull/141/commits/edf744a7e44bec23fd526bbda19c6c8b664b540f)